benchmark/path/basename.js

```diff
@@ -0,0 +1,26 @@
+var common = require('../common.js');
+var path = require('path');
+var v8 = require('v8');
+
+var bench = common.createBenchmark(main, {
+  type: ['win32', 'posix'],
+  n: [1e6],
+});
+
+function main(conf) {
+  var n = +conf.n;
+  var p = path[conf.type];
+
+  // Force optimization before starting the benchmark
+  p.basename('/foo/bar/baz/asdf/quux.html');
+  v8.setFlagsFromString('--allow_natives_syntax');
+  eval('%OptimizeFunctionOnNextCall(p.basename)');
+  p.basename('/foo/bar/baz/asdf/quux.html');
```

Comment:
This is because it takes V8 extra time to optimize the function when it is called after `%OptimizeFunctionOnNextCall` is called on it. So calling the function before the benchmark starts means that the time it takes for V8 to optimize the function does not get recorded as part of the benchmark time.